src/devices/QwiicButton/QwiicButton.QueueManipulation.cs

```diff
@@ -0,0 +1,119 @@
+﻿//// Licensed to the .NET Foundation under one or more agreements.
+//// The .NET Foundation licenses this file to you under the MIT license.
+//// See the LICENSE file in the project root for more information.
+
+using Iot.Device.QwiicButton.RegisterMapping;
+
+namespace Iot.Device.QwiicButton
+{
+    public sealed partial class QwiicButton
+    {
+        /// <summary>
+        /// Returns whether the queue of button press timestamps is full.
+        /// </summary>
+        public bool IsPressedQueueFull()
+        {
+            var pressedQueue = new QueueStatusBitField(_registerAccess.ReadSingleRegister(Register.PressedQueueStatus));
+            return pressedQueue.IsFull;
+        }
+
+        /// <summary>
+        /// Returns whether the queue of button press timestamps is empty.
+        /// </summary>
+        public bool IsPressedQueueEmpty()
+        {
+            var pressedQueue = new QueueStatusBitField(_registerAccess.ReadSingleRegister(Register.PressedQueueStatus));
+            return pressedQueue.IsEmpty;
+        }
+
+        /// <summary>
+        /// Returns how many milliseconds it has been since the last button press.
+        /// Since this returns a 32-bit unsigned int, it will roll over about every 50 days.
+        /// </summary>
+        public uint TimeSinceLastPress()
+        {
+            return _registerAccess.ReadQuadRegister(Register.PressedQueueFront);
+        }
+
+        /// <summary>
+        /// Returns how many milliseconds it has been since the first button press.
+        /// Since this returns a 32-bit unsigned int, it will roll over about every 50 days.
+        /// </summary>
+        public uint TimeSinceFirstPress()
+        {
+            return _registerAccess.ReadQuadRegister(Register.PressedQueueBack);
+        }
+
+        /// <summary>
+        /// Returns the oldest value in the queue (milliseconds since first button press), and then removes it.
+        /// </summary>
+        public uint PopPressedQueue()
+        {
+            var timeSinceFirstPress = TimeSinceFirstPress(); // Take the oldest value on the queue
+
+            var pressedQueue =
+                new QueueStatusBitField(_registerAccess.ReadSingleRegister(Register.PressedQueueStatus))
+                {
+                    PopRequest = true
+                };
+
+            // Remove the oldest value from the queue
+            _registerAccess.WriteSingleRegister(Register.PressedQueueStatus, pressedQueue.QueueStatusValue);
+
+            return timeSinceFirstPress; // Return the value we popped
+        }
+
+        /// <summary>
+        /// Returns whether the queue of button click timestamps is full.
+        /// </summary>
+        public bool IsClickedQueueFull()
```

Comment:
what's the difference between press and click? Is Click Down & Up and Press just down? Usual terminology for that is `KeyDown` and `Click`, I'd stick to that even if the sensor uses different naming in its docs

Reply:
I opted for `PressedDown` and `Clicked`. `KeyDown` is intended for keyboards. I also added some more explanation of the difference between PressedDown and Clicked.